resources/content_transform.go

```diff
@@ -32,29 +39,47 @@ func (h *MapEndpointHandler) ServeMapEndpoint(w http.ResponseWriter, r *http.Req
 }
 
 func (h *MapEndpointHandler) mapContent(w http.ResponseWriter, r *http.Request, transactionID string) {
-	methodePlaceholder, err := h.mapper.NewMethodeContentPlaceholderFromHTTPRequest(r)
-	uuid := methodePlaceholder.UUID
+	methodePlaceholder, err := h.NewMethodeContentPlaceholderFromHTTPRequest(r)
+
 	if err != nil {
-		writeError(w, err, transactionID, uuid, r.RequestURI)
+		writeError(w, err, transactionID, "", r.RequestURI)
 		return
 	}
-	upPlaceholder, err := h.mapper.MapContentPlaceholder(methodePlaceholder)
+	uuid := methodePlaceholder.UUID
+
+	uppPlaceholder, uppComplementaryContent, err := h.mapper.MapContentPlaceholder(methodePlaceholder)
 	if err != nil {
 		writeError(w, err, transactionID, uuid, r.RequestURI)
 		return
 	}
 
-	if upPlaceholder.IsMarkedDeleted {
+	if uppPlaceholder.IsMarkedDeleted {
 		writeMessageForDeletedContent(w, transactionID, uuid, r.RequestURI)
 		return
 	}
 
 	w.Header().Add("Content-Type", "application/json")
 	encoder := json.NewEncoder(w)
-	encoder.Encode(upPlaceholder)
+	encoder.Encode(h.createPublicationEventsMessages(uppPlaceholder, uppComplementaryContent))
 	log.WithField("transaction_id", transactionID).WithField("uuid", uuid).WithField("request_uri", r.RequestURI).Info("Transformation successful")
 }
 
+func (h *MapEndpointHandler) NewMethodeContentPlaceholderFromHTTPRequest(r *http.Request) (*model.MethodeContentPlaceholder, *utility.MappingError) {
```

Comment:
I think you should remove the receiver and move this function to the `methodeContentPlaceholder` file.